SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -28,169 +28,142 @@ void USpatialTypebindingManager::Init(USpatialNetDriver* InNetDriver)
-				}
-			}
+		// Make a copy of the already made FClassInfo for this specific subobject
+		FClassInfo SubobjectInfo = *SubobjectInfoPtr;
 
-			if (Property->PropertyFlags & CPF_AlwaysInterested)
-			{
-				for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
-				{
-					FInterestPropertyInfo InterestInfo;
-					InterestInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
-					InterestInfo.Property = Property;
-
-					Info.InterestProperties.Add(InterestInfo);
-				}
-			}
-		}
+		SubobjectInfo.SubobjectName = SubobjectSchemaData.Name;
 
 		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 		{
-			Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+			Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
 			if (ComponentId != 0)
 			{
-				Info.SchemaComponents[Type] = ComponentId;
-				ComponentToClassMap.Add(ComponentId, Class);
-				ComponentToOffsetMap.Add(ComponentId, 0);
+				SubobjectInfo.SchemaComponents[Type] = ComponentId;
+				ComponentToClassMap.Add(ComponentId, SubobjectClass);
+				ComponentToOffsetMap.Add(ComponentId, Offset);
 				ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
 			}
 		});
 
-		Info.Class = Class;
-
-		ClassInfoMap.Emplace(Class, Info);
+		Info.SubobjectInfo.Add(Offset, MakeShared<FClassInfo>(SubobjectInfo));
 	}
 
-	for (UClass* Class : SupportedClasses)
-	{
-		for (auto& SubobjectDataPair : SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SubobjectData)
-		{
-			int32 Offset = SubobjectDataPair.Key;
-			FSubobjectSchemaData SubobjectSchemaData = SubobjectDataPair.Value;
-
-			FClassInfo* ActorInfo = FindClassInfoByClass(Class);
-
-			FSoftClassPath SubobjectClassPath(SubobjectSchemaData.ClassPath);
-			UClass* SubobjectClass = SubobjectClassPath.TryLoadClass<UObject>();
-			if (SubobjectClass == nullptr)
-			{
-				continue;
-			}
-
-			FClassInfo* SubobjectInfoPtr = FindClassInfoByClass(SubobjectClass);
-			if (SubobjectInfoPtr == nullptr)
-			{
-				continue;
-			}
-
-			// Make a copy of the already made FClassInfo for this specific subobject
-			FClassInfo SubobjectInfo = *SubobjectInfoPtr;
-
-			SubobjectInfo.SubobjectName = SubobjectSchemaData.Name;
-
-			ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
-			{
-				Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
-				if (ComponentId != 0)
-				{
-					SubobjectInfo.SchemaComponents[Type] = ComponentId;
-					ComponentToClassMap.Add(ComponentId, SubobjectClass);
-					ComponentToOffsetMap.Add(ComponentId, Offset);
-					ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
-				}
-			});
-
-			ActorInfo->SubobjectInfo.Add(Offset, MakeShared<FClassInfo>(SubobjectInfo));
-		}
-	}
+	return Info;
 }
 
 FClassInfo* USpatialTypebindingManager::FindClassInfoByClass(UClass* Class)
 {
+	if (!ClassInfoMap.Contains(Class))
```

Comment:
I wonder if we should check whether the given class is also in the schema database class paths.

Reply:
Adding a IsSupportedClass() initially, logging the warning, and return nullptr sounds like a good idea.